jetty-util/src/main/java/org/eclipse/jetty/util/ssl/KeyStoreScanner.java

```diff
@@ -135,8 +135,7 @@ public void scan()
 
             _scanner.scan(callback);
             _scanner.scan(callback);
-            complete.await(10, TimeUnit.SECONDS);
-
+            return complete.await(10, TimeUnit.SECONDS);
```

Comment:
@sbordet this will return true even if the callback fails, which  is a bit strange.

Reply:
@gregw I don't understand? If the 10 s elapsed, `complete.await()` would have returned `false`, and the return value would have been ignored. Now we return it so that a JMX call to this method would know if it timed out.

Reply:
Scan is passed a callback. If it is failed then this method should return false.   Moreover I think if it times out then I think this method should throw a timeout exception rather than return

Reply:
@sbordet ^

Also I don't like the 10second wait either, but not sure how to avoid other than infinite wait